Scripts/Python/xRegisterAge.py

```diff
@@ -0,0 +1,120 @@
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+from PlasmaKITypes import *
+
+act = ptAttribActivator(1, "Act: Buttons")
+resp = ptAttribResponder(2, "Resp: Button Push")
+ageFileName = ptAttribString(3, "age filename")
+ageInstanceName = ptAttribString(4, "age instance name")
+
+class xRegisterAge(ptResponder):
+
+    def __init__(self):
+        # run parent class init
+        ptResponder.__init__(self)
+        self.id = 5008
+        version = 1
+        self.version = version
+        PtDebugPrint("__init__xRegisterAge v.", version)
+        
+    def OnNotify(self,state,id,events):
+        if state and PtFindAvatar(events) == PtGetLocalAvatar():
+            if id==act.id:
+                if not self.IFindAgeLinkInFolder(ageFileName.value):
+                    if resp.value:
+                        resp.run(self.key,events=events)
+                    PtSendKIMessage(kKILocalChatStatusMsg, PtGetLocalizedString("Global.Messages.AgeBookGet", [ageInstanceName.value]))
+                    self.CheckAndRegisterAge(ageFileName.value, ageInstanceName.value)
+
+    def IFindAgeLinkInFolder(self, ageName):
+        vault = ptVault()
+        folder = vault.getAgesIOwnFolder()
+        ageName = ageName.lower()
+        contents = folder.getChildNodeRefList()
+        for content in contents:
+            link = content.getChild().upcastToAgeLinkNode()
+            if link is not None:
+                info = link.getAgeInfo()
+            else:
+                link = content.getChild()
+                info = link.upcastToAgeInfoNode()
+
+            if info.getAgeFilename().lower() == ageName:
+                return link
+        return None
+
+    def CheckAndRegisterAge(self, ageFileName, ageInstanceName):
+        PtDebugPrint("xRegisterAge.CheckAndRegisterAge(): ",ageFileName,"; ",ageInstanceName)
+        vault = ptVault()
+        ageStruct = ptAgeInfoStruct()
+        ageStruct.setAgeFilename(ageFileName)
+        ageLinkNode = vault.getOwnedAgeLink(ageStruct)
+        if not ageLinkNode:
+            info = ptAgeInfoStruct()
+            info.setAgeFilename(ageFileName)
+            info.setAgeInstanceName(ageInstanceName)
+
+            playerName = PtGetClientName()
+            ageGuid = PtGuidGenerate()
+            userDefName = ""
+            desc = ""
+
+            if playerName[-1] == "s" or playerName[-1] == "S":
+                userDefName = "%s'" % playerName
+                desc = "%s' %s" % (playerName, info.getAgeInstanceName())
+            else:
+                userDefName = "%s's" % playerName
+                desc = "%s's %s" % (playerName, info.getAgeInstanceName())
```

Comment:
I think we have a localization tool to manage possessives now. Is that right, @ZarothYe?

Reply:
The user defined name is also set here, that function will work for the the  age description
Should I just leave the userDefName part in for another PR to have it covered by xLocTools?